ooniapi/services/ooniprobe/src/ooniprobe/routers/v1/probe_services.py

```diff
@@ -0,0 +1,126 @@
+
+    token = probe_login.username
+    # TODO: We have to find a way to explicitly log metrics with prometheus.
+    # We're currently using the instrumentator default metrics, like http response counts 
+    # Maybe using the same exporter as the instrumentator?
+    try:
+        dec = decode_jwt(token, audience="probe_login", key=settings.jwt_encryption_key)
+        registration_time = dec["iat"]
+        log.info("probe login: successful")
+        # metrics.incr("probe_login_successful")
+    except jwt.exceptions.MissingRequiredClaimError:
+        log.info("probe login: invalid or missing claim")
+        # metrics.incr("probe_login_failed")
+        raise HTTPException(status_code=401, detail="Invalid credentials") 
+    except jwt.exceptions.InvalidSignatureError:
+        log.info("probe login: invalid signature")
+        # metrics.incr("probe_login_failed")
+        raise HTTPException(status_code=401, detail="Invalid credentials") 
+    except jwt.exceptions.DecodeError:
+        # Not a JWT token: treat it as a "legacy" login
+        # return jerror("Invalid or missing credentials", code=401)
+        log.info("probe login: legacy login successful")
+        # metrics.incr("probe_legacy_login_successful")
+        registration_time = None
+
+    exp = datetime.now(timezone.utc) + timedelta(days=7)
+    payload = {"registration_time": registration_time, "aud": "probe_token"}
+    token = create_jwt(payload, key=settings.jwt_encryption_key)
+    # expiration string used by the probe e.g. 2006-01-02T15:04:05Z
+    expire = exp.strftime("%Y-%m-%dT%H:%M:%SZ")
+    login_response = ProbeLoginResponse(token=token, expire = expire)
+    setnocacheresponse(response)
+
+    return login_response
+
+class ProbeRegister(BaseModel):
+    # None of this values is actually used, but I add them
+    # to keep it compliant with the old api
+    password : str
+    platform : str
+    probe_asn : str
+    probe_cc : str
+    software_name : str 
+    software_version : str
+    supported_tests : List[str]
+
+class ProbeRegisterResponse(BaseModel):
+    client_id: str 
+
+@router.post("/register", tags=["ooniprobe"], response_model=ProbeRegisterResponse)
+def probe_register_post(
+    probe_register : ProbeRegister, 
+    response : Response,
+    settings : Settings = Depends(get_settings),
+) -> ProbeRegisterResponse:
+    """Probe Services: Register
+
+    Probes send a random string called password and receive a client_id
+
+    The client_id/password tuple is saved by the probe and long-lived
+
+    Note that most of the request body arguments are not actually 
+    used but are kept here to use the same API as the old version
+    """
+    global log
+
+    # client_id is a JWT token with "issued at" claim and
+    # "audience" claim. The "issued at" claim is rounded up.
+    issued_at = int(time.time())
+    payload = {"iat": issued_at, "aud": "probe_login"}
+    client_id = create_jwt(payload, key=settings.jwt_encryption_key)
```

Comment:
We should take note of the fact the jwt_encryption_key is being shared for between probes and user authenticating via email in the auth service.

I don't think there is necessary a problem, given that the `aud` field is different, but we should make sure we don't forget to always check `aud` filed to avoid authentication bypasses.